collector/src/main/java/com/bitmovin/analytics/retryBackend/RetryBackend.kt

```diff
@@ -0,0 +1,87 @@
+package com.bitmovin.analytics.retryBackend
+
+import android.os.Handler
+import android.os.SystemClock
+import android.util.Log
+import com.bitmovin.analytics.data.AdEventData
+import com.bitmovin.analytics.data.Backend
+import com.bitmovin.analytics.data.CallbackBackend
+import com.bitmovin.analytics.data.EventData
+import java.net.ConnectException
+import java.net.SocketTimeoutException
+import java.net.UnknownHostException
+import java.util.Date
+import kotlin.Exception
+import okhttp3.internal.http2.StreamResetException
+
+class RetryBackend(private val next: CallbackBackend, val handler: Handler) : Backend {
```

Comment:
maybe there is a better name for the `handler` field. I am not sure what this is actually doing but it looks like it's a common thing in Android.,

Reply:
renamed handler and runnable in https://github.com/bitmovin/bitmovin-analytics-collector-android/pull/132/commits/0b5f3e8dcbe9116a274fd705ccfda193d020ff56
are `scheduleSampleHandler` and `sendNextSampleRunnable` better?